frontend/pages/index.tsx

```diff
@@ -20,16 +19,18 @@ export default function Home({ contractName }: Props) {
       <main>
         <div className="relative flex min-h-screen flex-col justify-center overflow-hidden bg-gray-50 py-6 sm:py-12">
           <div className="absolute inset-0 [mask-image:linear-gradient(180deg,white,rgba(255,255,255,0))]"></div>
-          <ContractView contractName={contractName} />
+          <WalletSelectorContextProvider network={network} createAccessKeyFor={contractName}>
+            <ContractView contractName={contractName} />
+          </WalletSelectorContextProvider>
           <ExplainText />
         </div>
       </main>
-
     </>
   );
 }
 
 Home.getInitialProps = async () => {
-  const contractName = process.env.CONTRACT_NAME || 'guest-book.testnet';
-  return { contractName };
+  const contractName = process.env.NEXT_PUBLIC_CONTRACT_NAME || 'guest-book.testnet';
+  const network = process.env.NEXT_PUBLIC_NEAR_NETWORK === 'mainnet' ? 'mainnet' : 'testnet';
```

Comment:
As we have `.env` config file let's leave this:
```suggestion
  const contractName = process.env.NEXT_PUBLIC_CONTRACT_NAME;
  const network = process.env.NEXT_PUBLIC_NEAR_NETWORK;
```

Reply:
Sounds good to me! I added a default `NEAR_NETWORK` to `.env`.